### DISCUSSION TITLE
Need Help: Button Click Not Redirecting to New Page in Next.js

### DISCUSSION THREAD

**Discussion by krishp4telll** [vercel/next.js#74225] [Help]
### Summary

Hi everyone,

I’m new to Next.js, and I’m trying to set up a simple button that redirects to another page when clicked. I’ve created a pages folder with two files: index.js and about.js.

Below is the code I’m using in index.js:

The page loads fine, but when I click the button, it doesn’t navigate to /about. What am I doing wrong?

I’d appreciate any advice or examples of how to fix this. Thanks!

### Additional information

```js
import Link from 'next/link';  

export default function Home() {  
  return (  
    <div>  
      <h1>Welcome to my site</h1>  
      <button onClick={() => window.location.href = '/about'}>Go to About</button>  
    </div>  
  );  
}
```

**Comment by eric-mxrtin** [comment]


**Reply by krishp4telll** [reply]
thank you so much!

**Comment by icyJoseph** [comment]
Did you get help with this? You need to use the `useRouter` hook, which has push/replace methods.

While, doing `window.location.href` setting, or `window.location.replace(/* destination */)` (favoured method), are sort of correct, when using SPA's one wants to avoid unloading the document, and just tell the router to load the next given route.
